File: API/HandsOn/angular-with-api/src/app/shared/product.service.ts
import { Injectable } from '@angular/core';
import { Product } from '../models/product';
import { HttpClient } from '@angular/common/http';
import  { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ProductService {
api_url:string='http://localhost:5284/api/Product/';
  constructor(private http:HttpClient) { }
  GetAllProducts():Observable<Product[]> //Get all the products
  {
    return this.http.get<Product[]>(this.api_url+'GetAllProducts');
  }
  GetProduct(id:number):Observable<Product> //Get Product by id
  {
    return this.http.get<Product>(this.api_url+'GetProduct/'+id);
  }
  AddProduct(item:Product):Observable<any> //Add Product
  {
    return this.http.post(this.api_url+'AddProduct',item);
  }
  EditProduct(item:Product):Observable<any> //Edit Product
  {
    return this.http.put(this.api_url+'EditProduct',item);
  }
  DeleteProduct(id:number):Observable<any> //Delete Product
  {
    return this.http.delete(this.api_url+'DeleteProduct/'+id);
  }
}

File: API/HandsOn/HandsMVCWithAPI/Services/ProductService.cs
using HandsMVCWithAPI.Contracts;
using HandsMVCWithAPI.Models;
using Newtonsoft.Json;
using System.Net.Http;
using System.Net.Http.Headers;

namespace HandsMVCWithAPI.Services
{
    public class ProductService : IProductContract
    {
        private string api_url = "http://localhost:5284/";
        public  void AddProduct(Product product) //Add new Product
        {
            using (HttpClient client = new HttpClient())
            {
                client.BaseAddress = new Uri(api_url);
                //serialzie product details to json form.
                var contentData = new StringContent(JsonConvert.SerializeObject(product),
                    System.Text.Encoding.UTF8, "application/json");
                HttpResponseMessage response =  client.PostAsync("api/Product/AddProduct", contentData).Result;
                
            }
        }

        public void DeleteProduct(int id)
        {
            using (HttpClient client = new HttpClient())
            {
                client.BaseAddress = new Uri(api_url); //assign api addess 
              
              
                HttpResponseMessage response = client.DeleteAsync("api/Product/DeleteProduct/" + id).Result;
              
            }
        }

        public void EditProduct(Product product)
        {
            using (HttpClient client = new HttpClient())
            {
                client.BaseAddress = new Uri(api_url);
                //serialzie product details to json form.
                var contentData = new StringContent(JsonConvert.SerializeObject(product),
                    System.Text.Encoding.UTF8, "application/json");
                HttpResponseMessage response = client.PutAsync("api/Product/EditProduct", contentData).Result;

            }
        }

        public Product GetProduct(int id)
        {
            using (HttpClient client = new HttpClient())
            {
                client.BaseAddress = new Uri(api_url); //assign api addess 
                //set the media format to json
                MediaTypeWithQualityHeaderValue contentType = new MediaTypeWithQualityHeaderValue("application/json");
                client.DefaultRequestHeaders.Accept.Add(contentType);
                HttpResponseMessage response = client.GetAsync("api/Product/GetProduct/" + id).Result;
                Product product = JsonConvert.DeserializeObject<Product>(response.Content.ReadAsStringAsync().Result);
                return product;
            }
        }

        public List<Product> GetProducts() //Get All Products
        {
            using (HttpClient client = new HttpClient())
            {
                client.BaseAddress = new Uri(api_url); //assign api addess 
                //set the media format to json
                MediaTypeWithQualityHeaderValue contentType = new MediaTypeWithQualityHeaderValue("application/json");
                client.DefaultRequestHeaders.Accept.Add(contentType);
                HttpResponseMessage response = client.GetAsync("api/Product/GetAllProducts").Result;
                List<Product> list = JsonConvert.DeserializeObject<List<Product>>(response.Content.ReadAsStringAsync().Result);
                return list;
            }
        }
       
    }
}
